cookies/views/giles.py

```diff
@@ -222,15 +222,59 @@ def test_giles_cleanup(request):
 
 @login_required
 def log(request):
-    qs = auth.apply_filter(ResourceAuthorization.VIEW, request.user,
-                           GilesUpload.objects.all())
+    if request.method == 'GET':
+        qs = auth.apply_filter(ResourceAuthorization.VIEW, request.user,
+                            GilesUpload.objects.all())
+        filtered_objects = GilesUploadFilter(request.GET, queryset=qs)
+        context = {
+            'filtered_objects': filtered_objects,
+            'upload_form': GilesLogReuploadForm(),
+        }
+        return render(request, 'giles_log.html', context)
+    elif request.method == 'POST':
+        qs = auth.apply_filter(ResourceAuthorization.VIEW, request.user,
+                               GilesUpload.objects.all())
+        filtered_objects = GilesUploadFilter(request.GET, queryset=qs)
+        upload_form = GilesLogReuploadForm(request.POST, queryset=filtered_objects.qs)
+        if upload_form.is_valid():
+            upload_type = upload_form.cleaned_data.get('upload_type')
+            if upload_type == GilesLogReuploadForm.UPLOAD_ALL:
+                reupload = filtered_objects.qs
+            else:
+                reupload = upload_form.cleaned_data.get('reupload')
+            auth_reupload = auth.apply_filter(
+                ResourceAuthorization.EDIT,
+                request.user,
+                reupload)
+
+            context = {}
+            reuploads_count = reupload.count()
+            reuploads_success = 0
+            if auth_reupload.count() > 0:
+                reuploads_success = auth_reupload.update(state=GilesUpload.PENDING)
+                context.update({
+                    'reuploads_success': reuploads_success,
+                })
 
-    filtered_objects = GilesUploadFilter(request.GET, queryset=qs)
+            if reuploads_count > 0:
+                context.update({
+                    'reuploads_skipped': (reuploads_count - reuploads_success),
+                })
 
-    context = {
-        'filtered_objects': filtered_objects
-    }
-    return render(request, 'giles_log.html', context)
+            qs = auth.apply_filter(ResourceAuthorization.VIEW, request.user,
+                                   GilesUpload.objects.all())
+            filtered_objects = GilesUploadFilter(request.GET, queryset=qs)
+            context.update({
+                'filtered_objects': filtered_objects,
+                'upload_form': GilesLogReuploadForm(),
+            })
+            return render(request, 'giles_log.html', context)
+        else:
+            context = {
+                'filtered_objects': filtered_objects,
+                'upload_form': upload_form,
+            }
+            return render(request, 'giles_log.html', context)
```

Comment:
we probalby want to make sure that uploads that were successfully uploaded before, won't be uploaded again. otherwise we have two uploads in Giles for one resource in amphora.